util/signature/simple_hmac.go

```diff
@@ -105,41 +105,38 @@ func loadSigningKey(keyConfig string) (*common.Hash, error) {
 	return &hash, nil
 }
 
+func prependBytes(first []byte, rest ...[]byte) [][]byte {
+	return append([][]byte{first}, rest...)
+}
+
 // On success, extracts the message from the message+signature data passed in, and returns it
-func (h *SimpleHmac) VerifyMessageSignature(prefix []byte, data []byte) ([]byte, error) {
-	if len(data) < 32 {
-		return nil, errors.New("data is too short to contain message signature")
-	}
-	msg := data[32:]
+func (h *SimpleHmac) VerifySignature(sig []byte, data ...[]byte) (bool, error) {
```

Comment:
prependBytes has to create a new array every time.  It looks like Keccak256Hash() returns the same result if there are trailing `nil` parameters, so if you just had `VerifySignature(sig []byte, data1 []byte, data2 []byte), and the same for `SignSignature` you could remove the need for prependBytes.

Reply:
I think (not sure) that because keccak256 accepts a variable number of byte slices - calling it already creates an array of byte slices, so prependBytes should have a very minimal overhead. Even if it does create an extra array - it is a pointer array (slice of slices) that shouldn't copy actual data.
On the plus side - I like the interface that has variable number of byte slices.

Reply:
keccak256 actually loops through the slices one by one, so doesn't concatenate anything, which I like.  But your argument is sound :)